test/hotspot/gtest/runtime/test_os_windows.cpp

```diff
@@ -722,6 +722,101 @@ TEST_VM(os_windows, processor_count) {
+  }
+
+#if !defined(IA32)
+  if (isSupportedWindowsVersion && EnableAllLargePageSizes) {
+    size_t page_size_count = 0;
+    size_t page_size = os::page_sizes().largest();
+
+    do {
+      ++page_size_count;
+      page_size = os::page_sizes().next_smaller(page_size);
+    } while (page_size >= os::page_sizes().smallest());
+
+    EXPECT_GT(page_size_count, 1u) << "There should be multiple large page sizes available.";
+
+    size_t large_page_size = _large_page_size;
+    size_t min_size = os::page_sizes().smallest();
+
+    for (size_t page_size = os::page_sizes().largest(); page_size >= min_size; page_size = os::page_sizes().next_smaller(page_size)) {
+      EXPECT_TRUE(page_size % min_size == 0) << "Each page size should be a multiple of the minimum large page size.";
+      EXPECT_LE(page_size, large_page_size) << "Page size should not exceed the determined large page size.";
+    }
+  }
+#endif
+}
+
+TEST_VM(os_windows, large_page_init_decide_size) {
+  // Initial setup
+  UseLargePages = true;
+  bool isSupportedWindowsVersion = os::win32::is_windows_11_or_greater() ||
+    os::win32::is_windows_server_2022_or_greater();
+  LargePageSizeInBytes = 0; // Reset to default
+
+  // Test for large page support
+  size_t decided_size = os::large_page_init_decide_size();
+  size_t min_size = GetLargePageMinimum();
+  if (isSupportedWindowsVersion && min_size == 0) {
+    EXPECT_EQ(decided_size, 0) << "Expected decided size to be 0 when large page is not supported by the processor";
+  }
+
+  // Scenario 1: Test with 2MB large page size
+  LargePageSizeInBytes = 2 * M; // Set large page size to 2MB
+  decided_size = os::large_page_init_decide_size(); // Recalculate decided size
+  if (min_size == 2 * M) {
+    EXPECT_EQ(decided_size, 2 * M) << "Expected decided size to be 2M when large page and OS reported size are both 2M";
+  }
+
+  // Scenario 2: Test with 1MB large page size
+  LargePageSizeInBytes = 1 * M; // Set large page size to 1MB
+  decided_size = os::large_page_init_decide_size(); // Recalculate decided size
+  if (isSupportedWindowsVersion && min_size == 2 * M) {
+    EXPECT_EQ(decided_size, 2 * M) << "Expected decided size to be 2M when large page is 1M and OS reported size is 2M";
+  }
+
+  if (!isSupportedWindowsVersion) {
+#if defined(IA32) || defined(AMD64)
+    if (!EnableAllLargePageSizes && (min_size > 4 * M || LargePageSizeInBytes > 4 * M)) {
+      EXPECT_EQ(decided_size, 0) << "Expected decided size to be 0 for large pages bigger than 4mb on IA32 or AMD64";
+    }
+#endif
+  }
+
+  // Additional check for non-multiple of minimum size
+  LargePageSizeInBytes = 5 * min_size + 1; // Set an arbitrary large page size which is not a multiple of min_size
+  decided_size = os::large_page_init_decide_size(); // Recalculate decided size
+  if (LargePageSizeInBytes > 0 && LargePageSizeInBytes % min_size != 0) {
```

Comment:
Can `min_size` ever be 0? If so, we need to guard against doing a modulo operator with it. Also, given how LargePageSizeInBytes is set on line 813, are there any practical scenarios where the second part of the conditional would ever be false? If not, we can probably remove it.

Reply:
If the processor does not support large pages, the value can be 0. We need to guard it.Yes that is correct,it cant be false as both the conditions will be 1. We can remove it.